tests/test_dynamic_links.py

```diff
@@ -0,0 +1,212 @@
+                count=1)
+        assert 'not recognized' in str(excinfo.value)
+
+    @pytest.mark.parametrize('arg', dynamic_links.EventStats._platforms.keys())
+    def test_raw_platform_values_invalid(self, arg):
+        with pytest.raises(ValueError) as excinfo:
+            dynamic_links.EventStats(
+                platform=arg,
+                event=dynamic_links.EVENT_TYPE_CLICK,
+                count=1)
+        assert 'Raw string' in str(excinfo.value)
+
+    @pytest.mark.parametrize('event', dynamic_links.EventStats._event_types.keys())
+    def test_valid_event_values(self, event):
+        e_s = dynamic_links.EventStats(
+            platform=dynamic_links.PLATFORM_ANDROID,
+            event=dynamic_links.EventStats._event_types[event],
+            count=1)
+        assert e_s.event == dynamic_links.EventStats._event_types[event]
+
+    @pytest.mark.parametrize('arg', INVALID_STRINGS + ['unrecognized'])
+    def test_invalid_event_values(self, arg):
+        with pytest.raises(ValueError) as excinfo:
+            dynamic_links.EventStats(
+                platform=dynamic_links.PLATFORM_ANDROID,
+                event=arg,
+                count=1)
+        assert 'not recognized' in str(excinfo.value)
+
+    @pytest.mark.parametrize('arg', dynamic_links.EventStats._event_types.keys())
+    def test_raw_event_values_invalid(self, arg):
+        with pytest.raises(ValueError) as excinfo:
+            dynamic_links.EventStats(
+                platform=dynamic_links.PLATFORM_ANDROID,
+                event=arg,
+                count=1)
+        assert 'Raw string' in str(excinfo.value)
+
+    @pytest.mark.parametrize('count', [1, 123, 1234])
+    def test_valid_count_values(self, count):
+        e_s = dynamic_links.EventStats(
+            platform=dynamic_links.PLATFORM_ANDROID,
+            event=dynamic_links.EVENT_TYPE_CLICK,
+            count=count)
+        assert e_s.count == count
+
+    @pytest.mark.parametrize('arg', INVALID_NON_NEGATIVE_NUMS)
+    def test_invalid_count_values(self, arg):
+        with pytest.raises(ValueError) as excinfo:
+            dynamic_links.EventStats(
+                platform=dynamic_links.PLATFORM_ANDROID,
+                event=dynamic_links.EVENT_TYPE_CLICK,
+                count=arg)
+        assert 'must be a non negative int' in str(excinfo.value)
+
+
+class TestLinkStatsCreation(object):
+    @pytest.mark.parametrize('arg', INVALID_LISTS)
+    def test_invalid_event_stats_list(self, arg):
+        with pytest.raises(ValueError) as excinfo:
+            dynamic_links.LinkStats(arg)
+        assert'Must be a list or tuple' in str(excinfo.value)
+
+    @pytest.mark.parametrize('arg', [list([1, 2]), list('asdf'), tuple([1, 2])])
+    def test_empty_event_stats_list(self, arg):
+        with pytest.raises(ValueError) as excinfo:
+            dynamic_links.LinkStats(arg)
+        assert'elements of event stats must be "EventStats"' in str(excinfo.value)
+
+
+class TestQuoting(object):
```

Comment:
This feels redundant and out of place. You should just intercept the HTTP requests made by `get_stats()` tests and see if the request to backend is formatted correctly.